src/Tools/Source/RunTests/ProcessUtil.cs

```diff
@@ -10,9 +10,11 @@
 using System.Runtime.InteropServices;
 using System.Text;
 using System.Threading.Tasks;
+using System.Runtime.Versioning;
 
 namespace RunTests
 {
+    [SupportedOSPlatform("windows")]
```

Comment:
How does removing prebuilt detection force us to use this attribute?

Reply:
The change in the tool was to resolve analyzer warnings when SystemManagementVersion was upgraded from a 5.0 preview version to 7.0.0. the 7.0 assembly has proper OS support annotations:

https://github.com/dotnet/roslyn/pull/67645/files#diff-c77ad1cf833def4bef335849c7e456d0816d0f91af35dbffd26def0838c7e421R24

Reply:
We use RunTests on multiple OS though, can't make this a windows component. If we don't at the exact moment it's definitely the intent of the tool to be multi-platform.

Reply:
As far as my reading goes:

This API is not supported anywhere but Windows, and has attributes indicating such: `ManagementObject mo = new ManagementObject("win32_process.handle='" + p.Id + "'");` 

That means that `TryGetParentProcessId` will always return null (throw an unsupported exception). That in turn means that its caller `GetProcessChildrenCore` and `GetProcessTree` will always return empty lists.

Reply:
I can change to gate the use of ManagementObject on `if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))`, rather than using attributes. That would avoid changing the downstream code.